### PR TITLE
Add EPSG:7844 to fix ideal zoom for esri mapserver item.

### LINKED ISSUE #6590: "Ideal Zoom" broken on Esri MapServer when added through Add Web Data
Using e.g. the Transmission Lines layer in NatMap:

https://nationalmap.gov.au/#share=s-sSnPL1c2uOvNEW9F2eP7Xo50Vj

Clicking the Ideal Zoom button moves the map to the ideal zoom for the layer.

However, if you take the URL:

https://services.ga.gov.au/gis/rest/services/Foundation_Electricity_Infrastructure/MapServer

And add it through Add Web Data:

https://nationalmap.gov.au/#share=s-qocoeehV6CtYPlGwrsHVDZOHqvO

Clicking the Ideal Zoom button now does nothing, and no error or other message is logged in the console.

Testing this same thing with a WMS service, e.g.:

https://services.ga.gov.au/gis/services/Foundation_Electricity_Infrastructure/MapServer/WMSServer?request=GetCapabilities&service=WMS

When adding this through Add Web Data, adding the Transmission Lines layer and then clicking Ideal Zoom it works fine.

### PR DESCRIPTION
**Pull request by na9da**
### What this PR does

Fixes #6590 

- Adds proj4 definition for EPSG:7844 to Terria
  This is required to fix #6590.
- Converted modules `Proj4Defintions` and `Reproject` to Typescript.
- Enable spec for Reproject module.

### Test me

- Open this CI link for [main branch](http://ci.terria.io/main/#share=s-dpFrV1KOQnCzf8KVQBXQeSlfdu)
- Observe that the `Ideal zoom` workbench feature doesn't work for any of the dataset.
- Now open CI for [this branch](http://ci.terria.io/add-crs-7844/#share=s-dpFrV1KOQnCzf8KVQBXQeSlfdu).
- Observe that the `Ideal zoom` is now fixed.

### Checklist

- [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [ ] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Map/Vector/Proj4Definitions.ts [hunk 1]
```diff
@@ -60,7 +56,8 @@ var Proj4Definitions = {
   "EPSG:20258": "+proj=utm +zone=58 +south +ellps=aust_SA +units=m +no_defs",
   "EPSG:102100": "EPSG:3857",
   "EPSG:27700":
-    "+proj=tmerc +lat_0=49 +lon_0=-2 +k=0.9996012717 +x_0=400000 +y_0=-100000 +ellps=airy +towgs84=446.448,-125.157,542.06,0.15,0.247,0.842,-20.489 +units=m +no_defs"
+    "+proj=tmerc +lat_0=49 +lon_0=-2 +k=0.9996012717 +x_0=400000 +y_0=-100000 +ellps=airy +towgs84=446.448,-125.157,542.06,0.15,0.247,0.842,-20.489 +units=m +no_defs",
+  "EPSG:7844": "+proj=longlat +ellps=GRS80 +no_defs +type=crs"
```
Comment (na9da) [inline lib/Map/Vector/Proj4Definitions.ts:60]: New addition.

File: lib/Map/Vector/Reproject.ts
Comment (na9da) [inline lib/Map/Vector/Reproject.ts:1]: TSified.

File: lib/ModelMixins/GeojsonMixin.ts
Comment (na9da) [inline lib/ModelMixins/GeojsonMixin.ts:1]: Type fixes after TSifying `Reproject` module.

File: test/ViewModels/UploadDataTypesSpec.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import {
 } from "../../lib/Core/getDataType";
 import * as UploadDataTypes from "../../lib/ViewModels/UploadDataTypes";
 
-fdescribe("UploadDataTypes", function () {
+describe("UploadDataTypes", function () {
```
Comment (na9da) [inline test/ViewModels/UploadDataTypesSpec.ts:7]: Fix accidental commit from a previous PR.

File: lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts [hunk 3]
```diff
@@ -619,11 +615,11 @@ function getRectangleFromLayer(extent: Extent, rectangle: RectangleExtent) {
 
     const wkid = "EPSG:" + wkidCode;
 
-    if (!isDefined((proj4definitions as any)[wkid])) {
+    if (!isDefined((Proj4Definitions as any)[wkid])) {
```
Comment (mwu2018) [inline lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts]: `(Proj4Definitions as any)[wkid]` can be simplified as `Proj4Definitions[wkid]`. The same to the rest.

**Review by mwu2018 — CHANGES_REQUESTED** [review]
Looks good. Very minor changes might be required.

**Review by mwu2018 — APPROVED** [review]

**Pull request merged**